### PR TITLE
Enable Amazon Pay for use by US Single Contributors

### PR DESCRIPTION
**Pull request by JoemG**
## Why are you doing this?

PR to add the Login with Amazon and  Amazon Pay Wallet widget to the frontend and make the necessary calls to payment-api.

Login with Amazon docs:
https://developer.amazon.com/docs/login-with-amazon/javascript-sdk-reference.html#amazonlogin-methods

payment api pr
https://github.com/guardian/payment-api/pull/186

[**Trello Card**](https://trello.com)
https://trello.com/c/wiTUmfjd/1514-design-and-implement-amazon-pay

## Screenshots

Before selecting Amazon Pay:
Screenshot 2019-12-16 at 16 45 22

Before logging in with Amazon:
Screenshot 2019-12-16 at 16 45 29

Before clicking 'Contribute':
Screenshot 2019-12-16 at 16 45 36

### CODE REVIEW

File: support-frontend/assets/helpers/errorReasons.js [hunk 1]
```diff
@@ -27,15 +30,18 @@ function appropriateErrorMessage(errorReason: ?ErrorReason): ?string {
     case 'insufficient_funds':
       return 'The transaction was declined due to insufficient funds in your account. Please use a different card or contact your bank.';
     case 'payment_details_incorrect':
-      return 'An error occurred while trying to process your payment. Please double check your card details and try again. Alternatively, try another card or payment method.';
```
Comment (tomrf1) [inline support-frontend/assets/helpers/errorReasons.js]: this has changed the error messages for stripe as well, do we want to do this?

Comment (JoemG) [reply support-frontend/assets/helpers/errorReasons.js]: good spot... seem to have accidentally blatted that message completely :/

File: support-config/src/main/scala/com/gu/support/config/AmazonPayConfig.scala [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+class AmazonPayConfigProvider(config: Config, defaultStage: Stage) extends TouchpointConfigProvider[AmazonPayConfig](config, defaultStage) {
+  def fromConfig(config: Config): AmazonPayConfig = {
+    AmazonPayConfig(
+      config.getString("amazonpay.sellerId"),
+      config.getString("amazonpay.clientId")
+    )
+  }
+}
+
+//support/frontend/DEV/touchpoint.backend.environments.UAT.amazonpay.
```
Comment (jlieb10) [inline support-config/src/main/scala/com/gu/support/config/AmazonPayConfig.scala]: TBD: remove

File: support-frontend/app/config/Configuration.scala [hunk 3]
```diff
@@ -41,6 +41,8 @@ class Configuration(config: TypesafeConfig) {
 
   lazy val oneOffStripeConfigProvider = new StripeConfigProvider(config, stage, "oneOffStripe")
 
+  lazy val amazonPayConfigProvider = new AmazonPayConfigProvider(config, stage)
```
Comment (jlieb10) [inline support-frontend/app/config/Configuration.scala]: Check on other switch added by Paul

File: support-frontend/app/views/contributions.scala.html [hunk 4]
```diff
@@ -106,5 +114,6 @@
   window.guardian.stripeElementsRecurring = @settings.switches.experiments.get("stripeElementsRecurring").exists(_.isOn)
   window.guardian.forceCampaign = @settings.switches.experiments.get("forceCampaign").exists(_.isOn)
   window.guardian.recurringStripePaymentRequestButton = @settings.switches.experiments.get("recurringStripePaymentRequestButton").exists(_.isOn)
+
```
Comment (jlieb10) [inline support-frontend/app/views/contributions.scala.html]:
```suggestion
```

File: support-frontend/assets/helpers/contributions.js [hunk 5]
```diff
@@ -35,10 +36,11 @@ export const logInvalidCombination = (contributionType: ContributionType, paymen
   logException(`Invalid combination of contribution type ${contributionType} and payment method ${paymentMethod}`);
 };
 
+// Legacy type, only used by stripe checkout. Can be cleaned up after stripe checkout fully removed
```
Comment (jlieb10) [inline support-frontend/assets/helpers/contributions.js]: 👍 documentation

File: support-frontend/assets/pages/contributions-landing/components/AmazonPay/AmazonPayLoginButton.jsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      return (
+        <div>
+          <div id="AmazonLoginButton" />
+          <Button type="button" onclick={this.loginPopup(amazonLoginObject)} aria-label="Submit contribution">
+            Proceed with Amazon Pay
+          </Button>
+        </div>
+      );
+    }
+    // TODO - spinner?
```
Comment (jlieb10) [inline support-frontend/assets/pages/contributions-landing/components/AmazonPay/AmazonPayLoginButton.jsx]:
```suggestion
```

File: support-frontend/app/config/Configuration.scala (hunk 3 above)
Comment (tomrf1) [reply support-frontend/app/config/Configuration.scala]: admin-console PR merged, no further changes needed here

**Pull request merged**